src/Mutex.ts

```diff
@@ -10,6 +10,7 @@ class Mutex implements MutexInterface {
         const ticket = new Promise<MutexInterface.Releaser>(resolve => this._queue.push(resolve));
 
         if (!this._pending) {
+            this._pending = true;
```

Comment:
I would prefer if you moved this back to `_dispatchNext` --- the flag guards the execution of the next pending callback, and I feel it should be set as closely to the actual call location as possible.